clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -335,6 +335,14 @@ public void poll(long timeout) throws IOException {
         // we use the time at the end of select to ensure that we don't close any connections that
         // have just been processed in pollSelectionKeys
         maybeCloseOldestConnection(endSelect);
+
+        Iterator<String> connectedIt = connected.iterator();
+        while (connectedIt.hasNext()) {
+            if (disconnected.containsKey(connectedIt.next())) {
+                log.warn("Channel {} is marked connected and disconnected at the same time");
```

Comment:
Yeah. It is an internal class though. Users need to do nothing. I will update it.